node/core/approval-voting/src/lib.rs

```diff
@@ -72,32 +74,48 @@ mod import;
 mod time;
 mod persisted_entries;
 
+use crate::approval_db::v1::Config as DatabaseConfig;
+
 #[cfg(test)]
 mod tests;
 
 const APPROVAL_SESSIONS: SessionIndex = 6;
 const LOG_TARGET: &str = "parachain::approval-voting";
 
 /// Configuration for the approval voting subsystem
+#[derive(Debug, Clone)]
 pub struct Config {
-	/// The path where the approval-voting DB should be kept. This directory is completely removed when starting
-	/// the service.
-	pub path: std::path::PathBuf,
-	/// The cache size, in bytes, to spend on approval checking metadata.
-	pub cache_size: Option<usize>,
+	/// The column family in the DB where approval-voting data is stored.
+	pub col_data: u32,
 	/// The slot duration of the consensus algorithm, in milliseconds. Should be evenly
 	/// divisible by 500.
 	pub slot_duration_millis: u64,
 }
 
+// The mode of the approval voting subsystem. It should start in a `Syncing` mode when it first
+// starts, and then once it's reached the head of the chain it should move into the `Active` mode.
+//
+// In `Active` mode, the node is an active participant in the approvals protocol. When syncing,
+// the node follows the new incoming blocks and finalized number, but does not yet participate.
```

Comment:
so it can't transition from Active to Syncing? what if it's offline for a while?

Reply:
That almost never happens, so I haven't yet accounted for that case.